### PR TITLE
CBR/Backups: additional query opt to get resource by ID

### PR DESCRIPTION
**Pull request by artem-lifshits**
### What this PR does / why we need it
PR adds new list opt for provider [#1779](https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/issues/1779)

### CODE REVIEW

File: openstack/cbr/v3/backups/requests.go [hunk 1]
```diff
@@ -60,18 +57,70 @@ func (opts ListOpts) ToBackupListQuery() (string, error) {
...
+	var refinedBackups []Backup
+	var matched bool
+	m := map[string]interface{}{}
+	if opts.ID != "" {
+		m["ID"] = opts.ID
+	}
+
+	if opts.CheckpointID != "" {
+		m["CheckpointID"] = opts.CheckpointID
+	}
```
Comment (lego963) [inline openstack/cbr/v3/backups/requests.go]: Is this used in listOpts query or not?

File: openstack/cbr/v3/backups/requests.go [hunk 2]
```diff
@@ -60,18 +57,70 @@ func (opts ListOpts) ToBackupListQuery() (string, error) {
...
+	allBackups, err := ExtractBackups(pages)
+	if err != nil {
+		return nil, err
+	}
+
+	return FilterBackups(allBackups, opts)
+}
+
+func FilterBackups(backups []Backup, opts ListOpts) ([]Backup, error) {
+
```
Comment (lego963) [inline openstack/cbr/v3/backups/requests.go]: remove blank line

File: openstack/cbr/v3/backups/requests.go [hunk 3]
```diff
@@ -60,18 +57,70 @@ func (opts ListOpts) ToBackupListQuery() (string, error) {
...
+	q, err := golangsdk.BuildQueryString(&opts)
+	if err != nil {
+		return nil, err
+	}
+	url += q.String()
+
+	pages, err := pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
+		return BackupPage{pagination.LinkedPageBase{PageResult: r}}
+	}).AllPages()
+
```
Comment (lego963) [inline openstack/cbr/v3/backups/requests.go]: and here

File: openstack/cbr/v3/backups/results.go [hunk 4]
```diff
@@ -82,12 +82,28 @@ func (r GetResult) Extract() (*Backup, error) {
...
+
+func (r BackupPage) IsEmpty() (bool, error) {
+	is, err := ExtractBackups(r)
+	return len(is) == 0, err
 }
 
 func ExtractBackups(r pagination.Page) ([]Backup, error) {
 	var s []Backup
-	err := r.(BackupPage).Result.ExtractIntoSlicePtr(&s, "backups")
+	err := (r.(BackupPage)).Result.ExtractIntoSlicePtr(&s, "backups")
```
Comment (lego963) [inline openstack/cbr/v3/backups/results.go]: it can be simplified
```go
(r.(BackupPage)).ExtractIntoSlicePtr(&s, "backups")
```

**Review by lego963 — CHANGES_REQUESTED** [review]
can you add test with ID filtering?
like `TestVPCListing`

File: openstack/cbr/v3/backups/requests.go (hunk 1 above)
Comment (artem-lifshits) [reply openstack/cbr/v3/backups/requests.go]: fixed, thank you

**Review by lego963 — DISMISSED** [review]

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**